spec/ddtrace/contrib/http/request_spec.rb

```diff
@@ -424,22 +430,23 @@ def expect_request_without_distributed_headers
       let(:custom_error_message) { 'example error' }
       let(:custom_error) { StandardError.new(custom_error_message) }
       let(:span) { spans.first }
+      let(:expected_error) { "Failed to open TCP connection to #{host}:#{port} (#{custom_error_message})" }
 
-      before { stub_request(:get, "#{uri}#{path}").to_raise(custom_error) }
```

Comment:
I am not sure what was the real world use case for testing custom errors from `request`. According to the [documentation](https://ruby-doc.org/stdlib-2.7.0/libdoc/net/http/rdoc/Net/HTTP.html#method-i-request) it states this on the request methods:

> This method never raises Net::* exceptions.

I also tried to simulate some malformed HTTP response from the server but I did not manage to get exception.

So instead I've changed this to exception on socket open.

This is actually connected to the issue we are trying to solve here. Because the original `request` span was also encapsulating the connection opening, in real world those request spans could have exceptions. But since we split the connection spans out of the request spans, the request span will likely never have exceptions (at least if I get the documentation correctly)